modules/kubernetes-addons/helm_addon/main.tf

```diff
@@ -0,0 +1,71 @@
+resource "helm_release" "addon" {
+  count                      = var.manage_via_gitops ? 0 : 1
+  name                       = local.helm_config["name"]
+  repository                 = local.helm_config["repository"]
+  chart                      = local.helm_config["chart"]
+  version                    = local.helm_config["version"]
+  timeout                    = local.helm_config["timeout"]
+  values                     = local.helm_config["values"]
+  create_namespace           = local.helm_config["create_namespace"]
+  namespace                  = local.helm_config["namespace"]
+  lint                       = local.helm_config["lint"]
+  description                = local.helm_config["description"]
+  repository_key_file        = local.helm_config["repository_key_file"]
+  repository_cert_file       = local.helm_config["repository_cert_file"]
+  repository_ca_file         = local.helm_config["repository_ca_file"]
+  repository_username        = local.helm_config["repository_username"]
+  repository_password        = local.helm_config["repository_password"]
+  verify                     = local.helm_config["verify"]
+  keyring                    = local.helm_config["keyring"]
+  disable_webhooks           = local.helm_config["disable_webhooks"]
+  reuse_values               = local.helm_config["reuse_values"]
+  reset_values               = local.helm_config["reset_values"]
+  force_update               = local.helm_config["force_update"]
+  recreate_pods              = local.helm_config["recreate_pods"]
+  cleanup_on_fail            = local.helm_config["cleanup_on_fail"]
+  max_history                = local.helm_config["max_history"]
+  atomic                     = local.helm_config["atomic"]
+  skip_crds                  = local.helm_config["skip_crds"]
+  render_subchart_notes      = local.helm_config["render_subchart_notes"]
+  disable_openapi_validation = local.helm_config["disable_openapi_validation"]
+  wait                       = local.helm_config["wait"]
+  wait_for_jobs              = local.helm_config["wait_for_jobs"]
+  dependency_update          = local.helm_config["dependency_update"]
+  replace                    = local.helm_config["replace"]
+
+  postrender {
+    binary_path = local.helm_config["postrender"]
+  }
+
+  dynamic "set" {
+    iterator = each_item
+    for_each = local.helm_config["set"]
+
+    content {
+      name  = each_item.value.name
+      value = each_item.value.value
+    }
+  }
+
+  dynamic "set_sensitive" {
+    iterator = each_item
+    for_each = local.helm_config["set_sensitive"] == null ? [] : local.helm_config["set_sensitive"]
+
+    content {
+      name  = each_item.value.name
+      value = each_item.value.value
+    }
+  }
+  depends_on = [module.irsa]
+}
+
+module "irsa" {
+  source                            = "../../../modules/irsa"
+  eks_cluster_id                    = var.eks_cluster_id
+  create_kubernetes_namespace       = true
```

Comment:
the default helm config is true for creating a namespace as well. I'd suggest we prevent helm from creating the namespace and always do it explicitly through irsa?

Reply:
module also should be conditional as IRSA is not needed for all the add-ons